great_expectations/experimental/datasources/postgres_datasource.py

```diff
@@ -1,7 +1,16 @@
+from typing import TYPE_CHECKING
+
+from pydantic import constr
 from typing_extensions import Literal
 
 from great_expectations.experimental.datasources.sql_datasource import SQLDatasource
 
+if TYPE_CHECKING:
+    PostgresConnectionString = str
+else:
+    PostgresConnectionString = constr(regex=r"^postgresql")
+
 
 class PostgresDatasource(SQLDatasource):
     type: Literal["postgres"] = "postgres"  # type: ignore[assignment]
+    connection_string: PostgresConnectionString
```

Comment:
@billdirks did you try using the `PostgresDsn` type for this?
```python
from pydantic import PostgresDsn
```

```suggestion
    connection_string: PostgresDsn
```
https://docs.pydantic.dev/usage/types/#urls

Reply:
I hadn't tried this. Thanks for pointing it out! Done and I added tests.